src/App.tsx

```diff
@@ -1,61 +1,146 @@
+import { useState } from 'react';
+import { Todo } from './Types/Todo';
+import { User } from './Types/User';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const getUserById = (todo: Todo) => {
+  return usersFromServer.find(user => user.id === todo.userId) || null;
+};
+
+const getTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUserById(todo),
+}));
+
+const createTodo = (
+  selectedUser: User,
+  title: string,
+  todos: Todo[],
+): Todo => {
+  const lastId = Math.max(...todos.map(todo => todo.id));
+
+  return ({
+    id: lastId + 1,
+    title,
+    completed: false,
+    userId: selectedUser.id,
+    user: selectedUser,
+  });
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getTodos);
+  const [isUserError, setIsUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [isUserSelected, setIsUserSelected] = useState('0');
+
+  const handleInputTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    setTitleError(false);
+  };
+
+  const handleSelectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setIsUserSelected(event.target.value);
+
+    setIsUserError(false);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setIsUserSelected('0');
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const currentUser = usersFromServer
```

Comment:
reuse `getUserById` function

Reply:
why you igrone this?

Reply:
I spent a lot of time, and tried to do this with generics and union types, but it didn't work, always had some errors, don't know how to do this